benchmarks/triton_kernels_benchmark/benchmark_shapes_parser.py

```diff
@@ -0,0 +1,63 @@
+from typing import List, Union
+from dataclasses import dataclass, field
+
+import re
+
+
+@dataclass
+class ShapePatternParser:
+    pattern: str
+    pattern_tokens: List[str] = field(init=False)
+    pattern_dims: int = field(init=False)
+
+    def __post_init__(self):
+        self.pattern_tokens = self.parse(self.pattern, pattern_shape=True)
+        self.pattern_dims = len(self.pattern_tokens)
+
+    def __str__(self):
+        return self.pattern
+
+    @classmethod
+    def parse(cls, shape_string: str, pattern_shape: bool = False) -> List[Union[int, str]]:
```

Comment:
Class fields are not used here.
```suggestion
    @staticmethod
    def parse(shape_string: str, pattern_shape: bool = False) -> List[Union[int, str]]:
```

Reply:
I also assume by definition this method is private?